### ISSUE TITLE
Version 4.6.0 skips tests without apparent reason

### ISSUE THREAD

**Issue by Snawoot** [pytest-dev/pytest#5354]
Since version 4.6.0 pytest skips tests without apparent reason: https://travis-ci.org/Snawoot/postfix-mta-sts-resolver/jobs/540181138

- [x] output of `pip list` from the virtual environment you are using: **[HERE](https://travis-ci.org/Snawoot/postfix-mta-sts-resolver/jobs/540181138#L476)**
- [x] pytest and operating system versions: **pytest 4.6.0 on Ubuntu Xenial @ Travis CI**
- [x] minimal example if possible: **link above**

I can't understand why it happens, so I had to immediately rollback to 4.5.0 and fix this version in dev dependencies.

**Comment by asottile** [comment]
Can you add `-rs` (it should add additional reporting information about skipped tests)

**Comment by asottile** [comment]
This appears to be the minimal case to reproduce this:

```python
import itertools

import pytest

AS = (1, 2, 3)
BS = (4, 5, 6)


@pytest.mark.parametrize(('a', 'b'), itertools.product(AS, BS))
def test(a, b):
    pass
```

A workaround is to apply this diff:

```diff
-@pytest.mark.parametrize(('a', 'b'), itertools.product(AS, BS))
+@pytest.mark.parametrize(('a', 'b'), tuple(itertools.product(AS, BS)))
```

looking now to see what regressed this 🤔

**Comment by RonnyPfannschmidt** [comment]
There should be a warning

**Comment by asottile** [comment]
w/ `-rs` it produces this:

```console
$ pytest t.py -rs
============================= test session starts ==============================
platform linux -- Python 3.6.7, pytest-4.6.0, py-1.8.0, pluggy-0.12.0
rootdir: /home/asottile/workspace/pyupgrade
collected 1 item                                                               

t.py s                                                                   [100%]

=========================== short test summary info ============================
SKIPPED [1] t.py:9: got empty parameter set ('a', 'b'), function test at /home/asottile/workspace/pyupgrade/t.py:8
========================== 1 skipped in 0.01 seconds ===========================
```

**Comment by asottile** [comment]
Looks like this regressed in https://github.com/pytest-dev/pytest/pull/5254  CC @Sup3rGeo

**Comment by Snawoot** [comment]
@asottile Thank you! Workaround does just fine. Here is output with `-rs` option if still needed: https://travis-ci.org/Snawoot/postfix-mta-sts-resolver/jobs/540200985

**Comment by asottile** [comment]
#5356 has a fix for this -- thanks for the issue! 🎉

**Comment by asottile** [comment]
This has been [released as part of 4.6.1](https://twitter.com/codewithanthony/status/1135258518502514690) -- thanks again for the report!
